### PR TITLE
Fix #1, Fix #2 : add automated compilation and automated testing 

### LINKED ISSUE #1: Automated test execution
Property (core CI feature - testing): the CI server supports executing the automated tests of the group project. Testing is triggered as webhook, on the branch where the change has been made, as specified in the HTTP payload.

### LINKED ISSUE #2: Automated compilation
Property (core CI feature - compilation): the CI server supports compiling the group project, a static syntax check is to be performed for languages without compiler. Compilation is triggered as webhook, the CI server compiles the branch where the change has been made, as specified in the HTTP payload.

### PR DESCRIPTION
**Pull request by boesingerl**
Filled in the ContinuousIntegrationServer handle method, adding a reaction to the request :
Using the payload delivered by github, gets the url of the repo and the branch to clone (using GSON library)
From the url, clones the repo into the local working directory (using Jgit library)
From the cloned repo, builds the project and runs test (using GradleConnector library)

### CODE REVIEW

**Comment by boesingerl** [comment]
We should wait until we add tests to this branch before merging it to master.

File: src/main/java/ContinuousIntegrationServer.java [hunk 1]
```diff
@@ -2,44 +2,139 @@
...
+        outputStream.writeTo(new FileOutputStream(new File("log.txt")));
+        outputStream.close();
+        ps.close();
+
+        //SEND THE EMAIL
+        //author is the email adress, log is the string to send by email
     }
- 
+
+    private static void buildAndRun(String cloneDirectoryPath){
```
Comment (Pontus-Broberg) [inline src/main/java/ContinuousIntegrationServer.java]: Also add doc to this function.

File: src/main/java/ContinuousIntegrationServer.java [hunk 2]
```diff
@@ -2,44 +2,139 @@
...
+                    e.printStackTrace();
+                }
+            }).start();
+        }else{
+            response.getWriter().println("Payload is null");
+        }
+    }
 
-        response.getWriter().println("CI job done");
+    private void buildFromPayload(String payload) throws IOException {
```
Comment (Pontus-Broberg) [inline src/main/java/ContinuousIntegrationServer.java]: Would be nice to add doc to this function.

File: src/main/java/ContinuousIntegrationServer.java [hunk 3]
```diff
@@ -2,44 +2,139 @@
...
-public class ContinuousIntegrationServer extends AbstractHandler
-{
+import org.eclipse.jgit.api.errors.GitAPIException;
+
+import org.gradle.tooling.*;
+/**
+ * Skeleton of a ContinuousIntegrationServer which acts as webhook
+ * See the Jetty documentation for API documentation of those classes.
+ */
+public class ContinuousIntegrationServer extends AbstractHandler {
```
Comment (Pontus-Broberg) [inline src/main/java/ContinuousIntegrationServer.java]: Maybe change the doc for this function?

**Pull request merged**